templates/pkg/resource/sdk.go.tpl

```diff
@@ -182,19 +188,43 @@ func (rm *resourceManager) updateConditions (
 		awsErr, _ := ackerr.AWSError(err)
 		errorMessage := awsErr.Message()
 		terminalCondition.Message = &errorMessage
-	} else if terminalCondition != nil {
-		terminalCondition.Status = corev1.ConditionFalse
-		terminalCondition.Message = nil
+	} else {
+		// Clear the terminal condition if no longer present
+		if terminalCondition != nil {
+			terminalCondition.Status = corev1.ConditionFalse
+			terminalCondition.Message = nil
+		}
+		// Handling NonTerminal Conditions
+		if err != nil {
+			if nonTerminalCondition == nil {
+				// Add a new Condition containing a non-terminal error
+				nonTerminalCondition = &ackv1alpha1.Condition{
+					Type:   ackv1alpha1.ConditionTypeNonTerminal,
+				}
+				ko.Status.Conditions = append(ko.Status.Conditions, nonTerminalCondition)
+			}
+			nonTerminalCondition.Status = corev1.ConditionTrue
```

Comment:
`customUpdateConditions` should run before deciding if `Recoverable` can be set because one might set terminalCondition based on the error, it can be a service error or some other custom error